include/circt/Dialect/FIRRTL/FIRRTLOpInterfaces.td

```diff
@@ -127,7 +127,10 @@ def FModuleLike : OpInterface<"FModuleLike"> {
     InterfaceMethod<"Get a port's annotations attribute",
     "ArrayAttr", "getAnnotationsAttrForPort", (ins "size_t":$portIndex), [{}],
     /*defaultImplementation=*/[{
-      return $_op.getPortAnnotations()[portIndex].template cast<ArrayAttr>();
+      auto annos = $_op.getPortAnnotations();
+      if (annos.empty())
+        return ArrayAttr::get($_op.getContext(), {});
```

Comment:
Since `annos` is an empty array, you can return it instead of creating a new empty array, although it's a little confusing.

Reply:
😂 I didn't think of that! Pretty clever 😁. Would this make the code too hard to understand?

Reply:
You could add a comment? 

Empty arrays aren't expensive to create btw (at least when they have no explicit type), they are uniqued in the MLIRContext.

Reply:
Done!